main.py

```diff
@@ -0,0 +1,99 @@
+import streamlit as sl
+from transformers import pipeline, set_seed
+
+
+def is_int(str):
+    try:
+        int(str)
+        return True
+    except ValueError:
+        return False
+
+
+@sl.cache(suppress_st_warning=True, allow_output_mutation=True)
+def init():
+    return pipeline('text-generation', model='gpt2')
+
+
+GENERATOR = init()
```

Comment:
Making the model global is a bad solution. Since you are using a cache, you can call this function in the main method without any concerns.